src/core/lib/surface/init.cc

```diff
@@ -118,71 +120,85 @@ void grpc_init(void) {
   int i;
   gpr_once_init(&g_basic_init, do_basic_init);
 
-  gpr_mu_lock(&g_init_mu);
-  if (++g_initializations == 1) {
-    grpc_core::Fork::GlobalInit();
-    grpc_fork_handlers_auto_register();
-    gpr_time_init();
-    grpc_stats_init();
-    grpc_slice_intern_init();
-    grpc_mdctx_global_init();
-    grpc_channel_init_init();
-    grpc_core::ChannelzRegistry::Init();
-    grpc_security_pre_init();
-    grpc_core::ExecCtx::GlobalInit();
-    grpc_iomgr_init();
-    gpr_timers_global_init();
-    grpc_handshaker_factory_registry_init();
-    grpc_security_init();
-    for (i = 0; i < g_number_of_plugins; i++) {
-      if (g_all_of_the_plugins[i].init != nullptr) {
-        g_all_of_the_plugins[i].init();
+  int count;
+  do {
+    count = gpr_atm_no_barrier_load(&g_initializations);
```

Comment:
This is a standard double-checked-lock idiom.

DCL needs the fast path unlocked load to be acquire. (there are some alternatives if we care about non-x86 performance)